crates/bevy_ecs/src/system/query.rs

```diff
@@ -532,7 +532,32 @@ where
 
     /// Gets the query result for the given [`Entity`].
     ///
-    /// This can only be called for read-only queries, see [`Self::get_mut`] for write-queries.
+    /// In case of a nonexisting entity or mismatched component, a [`QueryEntityError`] is
+    /// returned instead.
+    ///
+    /// This can only be called for read-only queries (due to the [`ReadOnlyFetch`] trait bound).
+    /// see [`get_mut`](Self::get_mut) for queries that contain at least one mutable component.
+    ///
+    /// # Example
+    ///
+    /// In the following system, the `Entity` handle contained in the `target` resource is
+    /// used to get the `Person` component of that entity.
+    ///
+    /// ```
+    /// # use bevy_ecs::prelude::*;
+    /// #
+    /// # struct Person { name: String, credits: u32 }
+    /// # struct Target { entity: Entity }
```

Comment:
```suggestion
    /// # struct Winner { entity: Entity }
```

More descriptive names in examples help readers piece things together.